metrics/lib/src/metrics.dart

```diff
@@ -0,0 +1,139 @@
+import 'dart:io';
+import 'dart:convert';
+
+import 'package:crypto/crypto.dart';
+import 'package:json_annotation/json_annotation.dart';
+import 'package:metrics/src/target_os_type.dart';
+
+import 'version.dart';
+
+part 'metrics.g.dart';
+
+extension _IterableEx<T> on Iterable<T> {
+  T? get firstOrNull =>
+      cast<T?>().firstWhere((element) => true, orElse: () => null);
+}
+
+Future<Metrics> generateMetrics({
+  required Digest distinctId,
+  TargetOsType? targetOsType,
+  String? targetOsVersion,
+  Digest? anonymizedMacAddress,
+  Digest? anonymizedBundleId,
+}) async {
+  return Metrics(
+    event: 'run',
+    properties: Properties(
+      distinctId: distinctId,
+      token: 'ce0fac19508f6c8f20066d345d360fd0',
+      binding: 'dart',
+      language: 'dart',
+      framework: 'dart', // what about flutter?
```

Comment:
we could get if this is a flutter project from command line args. Ie have one here https://github.com/realm/realm-dart/pull/134/files#diff-72a322d353a66bf231af364f9d9c3958f4e6402d0c1e6d5ce8df34ad19e48073R8

Reply:
Check